### ISSUE TITLE
Statement is appended to previous line for some reason

### ISSUE THREAD

**Issue by jtojnar** [coccinelle/coccinelle#277]
Here is a simplified rule for converting function to return (void):

```smpl
@r1@
identifier func;
constant exit_val;
expression exit_val_preserved;
@@
void func(...) {
<+...
(
// Variant 1: a constant like a number or bool is returned.
// Let’s remove it without hesitation.
-return exit_val;
+return;
|
// Variant 2: a function is called in return statement.
// Let’s ensure it is run before returning.
-return exit_val_preserved;
+exit_val_preserved;
+return;
)
...+>
}
```

If I apply it onto the following code:

```c
void fun() {
  if (!test(model))
    return FALSE;

  return foo();
}
```

The inserted code attaches itself to the previous line for some reason.

```patch
--- test.c
+++ /tmp/cocci-output-1338955-01b7e5-test.c
@@ -1,6 +1,5 @@
 void fun() {
   if (!test(model))
-    return FALSE;
-
-  return foo();
+    return;foo();
+  return;
 }
```

I would expect something like:

```patch
@@ -1,6 +1,5 @@
 void fun() {
   if (!test(model))
-    return FALSE;
-
-  return foo();
+    return;
+  foo();
+  return;
 }
```

Reproducible on the [latest commit](https://github.com/coccinelle/coccinelle/commit/6d5602aca8775c3c5c503939c3dcf0637649d09b).

**Comment by JuliaLawall** [comment]
On Tue, 21 Dec 2021, Jan Tojnar wrote:

>
> Here is a simplified rule for converting function to return (void):

Thanks for the test case.  I will look into it.

julia

**Comment by jtojnar** [comment]
Looks like it behaves as expected on ac9af0e8b7c76341de1999389ce285b72ebaf81b:

```patch
--- foo.c
+++ /tmp/cocci-output-37849-bdb5b7-foo.c
@@ -1,6 +1,7 @@
 void fun() {
   if (!test(model))
-    return FALSE;
+    return;
 
-  return foo();
+  foo();
+  return;
 }
```

According to bisect, it was fixed in fad3acd364503a96292658b662c453398e8c6cde
